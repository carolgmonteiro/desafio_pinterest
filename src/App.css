@import url("https://fonts.googleapis.com/css?family=Roboto:400,500,700,900&display=swap");
p,
h1,
h2,
h3,
h4,
h5,
h6 {
  font-family: "Roboto", sans-serif;
}
.my-masonry-grid {
  display: -webkit-box; /* Not needed if autoprefixing */
  display: -ms-flexbox; /* Not needed if autoprefixing */
  display: flex;
  margin-left: -30px;
  /* gutter size offset */
  width: auto;
}
.my-masonry-grid_column {
  padding-left: 30px;
  /* gutter size */
  background-clip: padding-box;
}
.my-masonry-grid_column > div {
  /* change div to reference your elements you put in <Masonry> */
  margin-bottom: 10px;
}
.images-container {
  justify-content: center;
  margin-top: 18vh;
  padding-left: 5%;
  padding-right: 2%;
}

/* Optional, different gutter size on mobile */
@media (max-width: 800px) {
  .my-masonry-grid {
    margin-left: -15px; /* gutter size offset */
  }
  .my-masonry-grid_column {
    padding-left: 15px; /* gutter size offset */
  }
  .my-masonry-grid_column > div {
    margin-bottom: 15px; /* space between items */
  }
}
