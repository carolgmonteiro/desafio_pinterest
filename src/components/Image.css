.image-container {
  width: 220px;
  padding-bottom: 42px;
  padding-top: 8px;
  padding-left: 8px;
  padding-right: 8px;
  border-radius: 10px;
  position: relative;
  justify-content: center;
}
.image-container:hover {
  background: #efefef;
}

.linked-img {
  border-radius: 10px;
  width: 100%;
}

.linked-img:hover {
  cursor: zoom-in;
  filter: brightness(0.7);
}

.btn-save-collection {
  display: flex;
  flex-wrap: wrap;
}

.btn-save-collection button {
  height: 40px;
  border-top-right-radius: 8px;
  border-bottom-right-radius: 8px;
  font-weight: bold;
  width: 72px;
  border: none;
  color: white;
  background: #e60023;
  position: relative;
  justify-content: center;
  outline: none;
}

.btn-save-collection button:hover {
  background: #cf001f;
}

.btn-save-collection select {
  border-top-left-radius: 8px;
  border-bottom-left-radius: 8px;
  height: 40px;
  width: 128px;
  background: #efefef;
  border: 0;
  font-weight: 600;
  padding: 0;
  cursor: pointer;
  outline: none;
  scrollbar-arrow-color: #333333;
  border-color: transparent transparent rgba(0, 0, 0, 0.1) transparent;
}
.btn-save-collection select :hover {
  background: #e2e2e2;
}
.btn-dots {
  position: absolute;
  height: 20px;
  width: 20px;
  right: 15px;
}

.btn-dots button {
  border: none;
  background: transparent;
  height: 24px;
  outline: none;
  border-radius: 50%;
}
.btn-dots img:hover {
  background: #e4e3e3;
  border-radius: 50%;
}

.btn-dots img {
  right: 10px;
  bottom: 8px;
  cursor: pointer;
  height: 25px;
  width: 25px;
  border-radius: 50%;
  padding: 5px;
}
.link-content {
  display: flex;
  flex-wrap: wrap;
  /* position: absolute;
  bottom: 50px; */
}
.btn-share {
  border-radius: 50%;
  height: 30px;
  width: 30px;
  background: #474747;
  justify-content: center;
  right: 15px;
  cursor: pointer;
  position: absolute;
  bottom: 50px;
}
.btn-share img {
  width: 100%;
}

.btn-share:hover {
  background: #333333;
}

.link-image a {
  padding: 5px;
  border-radius: 20px;
  color: white;
  height: 30px;
  font-weight: bold;
  background: #474747;
  justify-content: center;
  right: 100px;
  cursor: pointer;
  border: none;
  font-size: 14px;
  position: absolute;
  bottom: 48px;
}

.link-image button:hover {
  background: #333333;
}
