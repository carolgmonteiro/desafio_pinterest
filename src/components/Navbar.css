#navbar {
  /* display: flex; */
  width: 100%;
  height: 65px;
  background: white;
  flex-wrap: wrap;
}

.header-content {
  z-index: 9999;
  top: 0;
  position: fixed;
  z-index: 9999;
  box-sizing: border-box;
}
.btn-box {
  display: flex;

  /* justify-content: space-around; */
}

#btn-pinterest img {
  height: 28px;
  width: 28px;
}

.logo-pinterest {
  border-radius: 50%;
  margin-top: 3px;
  margin-left: 15px;
  border: none;
  background: transparent;
}

.search-box-content {
  flex-basis: 65%;
  display: flex;
}

.icon {
  justify-content: center;
  /* padding: 2%; */
  padding: 10px;
  height: 48px;
  /* margin-right: 15px; */
  width: 48px;
  display: flex;
}

.button-content {
  /* margin-top: 5px; */
  flex-basis: 25%;
  display: flex;
  align-content: space-between;
}

.border {
  height: 50px;
  width: 2px;
  border-left: 2px solid #efefef;
  box-sizing: border-box;
  margin-left: 16px;
  display: block;
  margin-top: 15px;
}
hr {
  background: #8e8e8e;
  width: 100%;
  border-top: 1px solid #efefef !important;
  display: block;
  z-index: 9999;
  margin-top: 0 !important;
  margin-bottom: 0 !important;
  border-bottom: 0;
  border-left: 0;
  justify-content: flex-end;
}

/* .btn-nav-name {
  color: #8e8e8e;
  margin-bottom: 10px;
  margin-left: 10px;
  display: flex;
  flex-basis: 8%;
  background: white;
  font-weight: bold;
  font-size: 16px;
  border-radius: 20px;
  padding-left: 5px;
  padding-right: 5px;
  padding-top: 8px;
  padding-bottom: 8px;
  border: none;
}
.btn-nav-name :hover {
  background: #f2f2f2;
  cursor: pointer;
  height: 30px;
}

.user-icon {
  margin-right: 6px;
  height: 24px;
  width: 24px;
  border-radius: 50%;
  background: #8e8e8e;
  color: white;
  box-sizing: border-box;
} */
/* .btn-nav {
  color: #8e8e8e;
  margin-bottom: 10px;
  margin-left: 10px;
  display: flex;
  flex-basis: 8%;
  background: transparent;
  font-weight: bold;
  font-size: 16px;
  border-radius: 20px;
  padding-left: 5px;
  padding-right: 5px;
  padding-top: 8px;
  padding-bottom: 8px;
  border: none;
  cursor: pointer;
}
.btn-nav :hover {
  background: #f2f2f2;
  border-radius: 50%;
} */
