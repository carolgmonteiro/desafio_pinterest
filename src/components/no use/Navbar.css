/* NavBar */
#navbar {
  /* display: flex; */
  width: 100%;
  height: 65px;
  background: white;
  flex-wrap: wrap;
  z-index: 9999;
}

.header-content {
  z-index: 9999;
  top: 0;
  position: fixed;
  box-sizing: border-box;
}
.btn-box {
  display: flex;
  background: white;

  /* justify-content: space-around; */
}

#btn-pinterest img {
  height: 28px;
  width: 28px;
}
.logo-pinterest img:hover {
  background: #efefef;
  border-radius: 50%;
}

.logo-pinterest {
  margin-top: 3px;
  margin-left: 15px;
  border: none;
  background: transparent;
}

.search-box-content {
  flex-basis: 100%;
  display: flex;
}

.button-content {
  /* margin-top: 5px; */
  flex-basis: 25%;
  display: flex;
  align-content: space-between;
}

.border {
  height: 50px;
  width: 2px;
  border-left: 2px solid #efefef;
  box-sizing: border-box;
  margin-left: 16px;
  display: block;
  margin-top: 15px;
}
hr {
  background: #8e8e8e;
  width: 100%;
  border-top: 1px solid #efefef !important;
  display: block;
  z-index: 9999;
  margin-top: 0 !important;
  margin-bottom: 0 !important;
  border-bottom: 0;
  border-left: 0;
  justify-content: flex-end;
}

/* Search box */
.search-box {
  margin-top: 20px;
  flex-basis: 100%;
  display: flex;
}

.btn-search {
  border-top-left-radius: 8px;
  border-bottom-left-radius: 8px;
  color: #8e8e8e;
  padding-top: 6px;
  padding-left: 10px;
  margin-bottom: 5px;
  margin-left: 10px;
  display: flex;
  flex-basis: 15%;
  background: #efefef;
  font-weight: bold;
  font-size: 16px;
  border: none;
  height: 75%;
  width: 100%;
  cursor: pointer;
}

.search-box img {
  padding: 8px;
  margin-bottom: 15px;
  justify-content: center;
  margin-bottom: 15px;
  width: 38px;
  height: 38px;
  display: flex;
}

.search-box input {
  border-top-right-radius: 8px;
  border-bottom-right-radius: 8px;
  height: 75%;
  width: 100%;
  background: #efefef;
  border: 0;
  font-weight: 600;
  padding: 0;
  cursor: text;
}

/* Bton NavBar */
.button-box {
  margin-top: 20px;
  display: flex;
  align-content: space-between;
}

.btn-nav-name {
  height: 50px;
  width: 100%;
  color: #8e8e8e;
  /* margin-bottom: 10px; */
  margin-left: 5px;
  display: flex;
  flex-basis: 8%;
  background-color: white;
  font-weight: bold;
  font-size: 16px;
  border-radius: 25px;
  padding-left: 15px;
  padding-right: 15px;
  /* padding-top: 8px; */
  /* padding-bottom: 8px; */
  border: none;
  transition: 0.3s;
}
.btn-nav-name:hover {
  background-color: #f2f2f2;
  cursor: pointer;
  height: 50px;
  width: 100%;
}

.user-icon {
  margin-top: 13px;
  margin-right: 6px;
  height: 24px;
  width: 24px;
  border-radius: 50%;
  background: #8e8e8e;
  color: white;
  box-sizing: border-box;
}

.username {
  margin-top: 13px;
}

/* Btn NavBarIcon */

.btn-nav {
  color: #8e8e8e;
  margin-left: 10px;
  display: flex;
  flex-basis: 8%;
  background: transparent;
  border-radius: 20px;
  border: none;
  cursor: pointer;
}
.btn-nav :hover {
  background: #f2f2f2;
  border-radius: 50%;
}
.btn-nav button {
  outline: none;
}

.icon {
  justify-content: center;
  /* padding: 2%; */
  padding: 10px;
  height: 48px;
  margin-right: 2px;
  width: 48px;
  display: flex;
}

.btn {
  display: inline-block;
  font-weight: 400;
  color: #212529;
  text-align: center;
  vertical-align: middle;
  -webkit-user-select: none;
  user-select: none;
  background-color: transparent;
  border: 1px solid transparent;
  /* padding: 0.375rem 0.75rem; */
  font-size: 1rem;
  line-height: 1.5;
  border-radius: 0.25rem;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out,
    border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}
