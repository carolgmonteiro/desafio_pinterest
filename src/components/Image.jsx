import React from "react";
import "./Image.css";

class Image extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      hover: false
    };
  }

  handleHover() {
    this.setState({ hover: !this.state.hover });
  }

  render() {
    var imgStyle;
    var otherImgStyle;
    if (this.state.hover) {
      imgStyle = {
        display: "flex",
        position: "absolute",
        top: "10px",
        right: "10px"
      };
      otherImgStyle = { display: "block" };
    } else {
      imgStyle = { display: "none" };
      otherImgStyle = { display: "none" };
    }

    return (
      <React.Fragment>
        <div
          onMouseEnter={() => this.handleHover()}
          onMouseLeave={() => this.handleHover()}
          className="image-container"
        >
          <img
            className="linked-img"
            src={this.props.url}
            alt="link"
            data-target={"#modal" + this.props.dataTarget}
            data-toggle="modal"
          />
          <div style={imgStyle} className="btn-save-collection">
            <select name="collections">
              <option>illustration</option>
              <option>watercolor</option>
              <option>photos</option>
              <option>inspiration</option>
            </select>
            <button>Save</button>
          </div>
          <div className="btn-dots">
            <button>
              <img src={require("../img/dots.svg")} alt="dots" />
            </button>
          </div>
          <div className="link-content" style={otherImgStyle}>
            <div className="btn-share">
              <img src={require("../img/share.png")} alt="share" />
            </div>
            <div className="link-image">
              <a href={this.props.pageurl} target={"_blank"}>
                {this.props.pageurl.substr(27, 10)}...
              </a>
            </div>
          </div>
        </div>
      </React.Fragment>
    );
  }
}

export default Image;
