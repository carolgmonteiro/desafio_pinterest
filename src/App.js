import React from "react";
import Navbar from "./components/Navbar";
import "./App.css";
import InfiniteScroll from "react-infinite-scroll-component";
import Masonry from "react-masonry-css";
import Image from "./components/Image";
import Modal from "./components/Modal";
import Footer from "./components/Footer";

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      images: [],
      currentPage: 1,
      search: ""
    };
    // this.clickSearch = this.clickSearch.bind(this);
    // this.searchValue = this.searchValue.bind(this);
    this.fetchData = this.fetchData.bind(this);
    this.showImg = this.showImg.bind(this);
    this.showModal = this.showModal.bind(this);
  }

  componentDidMount() {
    fetch(
      "https://pixabay.com/api/?key=13902902-ce7912fe8b458917f397c8a5d&image_type=all&orientation=vertical&per_page=20&safesearch=true"
    )
      .then(response => response.json())
      .then(data => {
        return this.setState({ images: data.hits });
      })
      .catch(error => console.log("parsing failed", error));
  }

  fetchData() {
    const { currentPage } = this.state;
    this.setState({ currentPage: this.state.currentPage + 1 });
    fetch(
      `https://pixabay.com/api/?key=13902902-ce7912fe8b458917f397c8a5d&image_type=all&orientation=vertical&page=${currentPage}&per_page=20&safesearch=true`
    )
      .then(response => response.json())
      .then(data =>
        this.setState({ images: this.state.images.concat(data.hits) })
      )
      .catch(error => console.log("parsing failed", error));
  }

  showImg() {
    return this.state.images.map((image, i) => {
      return (
        <Image
          dataTarget={image.id}
          key={i}
          url={image.webformatURL}
          pageurl={image.pageURL}
        />
      );
    });
  }

  showModal() {
    return this.state.images.map((image, i) => {
      // console.log(image);
      return (
        <Modal
          id={image.id}
          key={i}
          url={image.webformatURL}
          pageurl={image.pageURL}
          tags={image.tags}
        />
      );
    });
  }

  // //guardar el valor del form
  // searchValue=(e)=> {
  //   this.setState({ search: e.target.value });
  //   console.log({ search: e.target.value });
  // }

  // //para buscar
  // clickSearch= (e) => {
  //   // e.preventDefault();
  //   if (e.key === "Enter") {
  //     this.searchImg(this.state.search);
  //     console.log("enter key pressed", this.state.search);
  //   }
  // }

  // searchImg(search) {
  //   fetch(
  //     `https://pixabay.com/api/?key=13902902-ce7912fe8b458917f397c8a5d&q=${search}image_type=all&orientation=vertical&per_page=20`
  //   )
  //     .then(response => response.json())
  //     .then(data => {
  //       return this.setState({ images: data.hits });
  //     })
  //     .catch(error => console.log("parsing failed", error));
  // }

  render() {
    const breakpointColumnsObj = {
      default: 5,
      1100: 4,
      700: 2,
      500: 1
    };

    return (
      <React.Fragment>
        <nav id="navbar" className="header-content">
          <Navbar
          // searchValue={this.searchValue}
          // clickSearch={this.clickSearch}
          />
          <hr />
        </nav>
        {this.showModal()}
        <div className="images-container">
          <InfiniteScroll
            dataLength={this.state.images.length}
            next={this.fetchData}
            hasMore={true}
            loader={<h4>Loading...</h4>}
            endMessage={
              <p style={{ textAlign: "center" }}>
                <b>Yay! You have seen it all</b>
              </p>
            }
          >
            <Masonry
              breakpointCols={breakpointColumnsObj}
              className="my-masonry-grid"
              columnClassName="my-masonry-grid_column"
            >
              {this.showImg()}
            </Masonry>
          </InfiniteScroll>
        </div>
        <Footer />
      </React.Fragment>
    );
  }
}

export default App;
