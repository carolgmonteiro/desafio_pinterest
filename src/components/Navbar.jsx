import React from "react";
import "./Navbar.css";
import axios from "axios";
import Image from "./Image";
import SearchBox from "./SearchBox";
// import BtnNavbar from "./BtnNavbar";
// import BtnNavbarIcon from "./BtnNavbarIcon";

class Navbar extends React.Component {
  state = {
    searchText: "",
    amount: 20,
    apiUrl: "https://pixabay.com/api",
    apiKey: "13902902-ce7912fe8b458917f397c8a5d",
    images: []
  };
  // onTextChange = e => {
  //   const valueSearch = e.target.value;
  //   this.setState({ [e.target.name]: valueSearch }, () => {
  //     if (valueSearch === " ") {
  //       this.setState({ images: [] });
  //     } else {
  //       axios
  //         .get(
  //           `${this.state.apiUrl}/?key=${this.state.apiKey}&q=${this.state.searchText}&image_type=all&orientation=vertical&per_page=${this.state.amount}&safesearch=true`
  //         )
  //         .then(res => this.setState({ images: res.data.hits }))
  //         .catch(err => console.log(err));
  //     }
  //   });
  // };
  //guardar el valor del form
  onTextChange = e => {
    this.setState({ [e.target.name]: e.target.value });
    // const valueSearch = e.target.value;
    // this.setState({ [e.target.name]: valueSearch });
    console.log(e.target.value);
  };

  //para buscar
  handleKeyPress = e => {
    e.preventDefault();
    if (e === " ") {
      this.setState({ images: [] });
    } else {
      axios
        .get(
          `${this.state.apiUrl}/?key=${this.state.apiKey}&q=${this.state.searchText}&image_type=all&orientation=vertical&per_page=${this.state.amount}&safesearch=true`
        )
        .then(res => this.setState({ images: res.data.hits }))
        .catch(err => console.log(err));
    }

    // if (e.key === "Enter") {
    //   this.searchImg(this.state.searchText);
    //   console.log("enter key pressed", this.state.searchText);
    // }
  };

  // searchImg = searchText => {
  //   axios
  //     .get(
  //       `${this.state.apiUrl}/?key=${this.state.apiKey}&q=${this.state.searchText}&image_type=all&orientation=vertical&per_page=${this.state.amount}&safesearch=true`
  //     )
  //     .then(res => res.json())
  //     .then(data => {
  //       return this.setState({ images: data.hits });
  //     })
  //     .catch(err => console.log(err));
  // };

  // searchImg(searchText) {
  //   fetch(
  //     `https://pixabay.com/api/?key=13902902-ce7912fe8b458917f397c8a5d&q=${searchText}image_type=all&orientation=vertical&per_page=20`
  //   )
  //     .then(response => response.json())
  //     .then(data => {
  //       return this.setState({ images: data.hits });
  //     })
  //     .catch(error => console.log("parsing failed", error));
  // }

  render() {
    console.log(this.state.images);
    return (
      <div className="btn-box">
        <button id="btn-pinterest" className="logo-pinterest">
          <img src={require("../img/logo.svg")} alt="logo-pinterest" />
        </button>
        <div className="search-box-content">
          <SearchBox />
          {/* <form className="search-box" onSubmit={this.handleKeyPress}>
            <button className="btn-search">
              <img src={require("../img/search.svg")} alt="search-button" />
            </button>
            <input
              type="search"
              placeholder="Search"
              name="searchText"
              value={this.state.searchText}
              onChange={this.onTextChange}
            />
            {this.state.images.length > 0 ? (
              <Image images={this.state.images} />
            ) : null}
          </form> */}
          <div className="button-content">
            <div className="button-box">
              <button className="btn-nav-name">Home</button>
              <button className="btn-nav-name">Following</button>
              <button className="btn-nav-name">
                <p className="user-icon">C</p>
                <p className="username">Carolina</p>
              </button>
            </div>
            <div className="border"></div>
            <button className="btn-nav">
              <img
                className="icon"
                src={require("../img/chat.svg")}
                alt="chat-button"
              />
            </button>

            <button className="btn-nav">
              <img
                className="icon"
                src={require("../img/bell.svg")}
                alt="notifications-button"
              />
            </button>
            <button className="btn-nav">
              <img
                className="icon"
                src={require("../img/dots.svg")}
                alt="settings-button"
              />
            </button>
          </div>
        </div>
      </div>
    );
  }
}

export default Navbar;

// class Navbar extends React.Component {
//   render() {
//     return (
//       <React.Fragment>
//         <nav id="navbar" className="header-content">
//           <div className="btn-box">
//             <button id="btn-pinterest" className="logo-pinterest">
//               <img src={require("../img/logo.svg")} alt="logo-pinterest" />
//             </button>
//             <div className="search-box-content">
//               <SearchBox />
//             </div>
//             <div className="button-content">
//               <BtnNavbar />
//               <div className="border"></div>
//               <BtnNavbarIcon />
//             </div>
//           </div>
//           <hr />
//         </nav>
//       </React.Fragment>
//     );
//   }
// }

// export default Navbar;
