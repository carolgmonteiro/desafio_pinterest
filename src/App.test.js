import React from "react";
import ReactDOM from "react-dom";
import App from "./App";
import Navbar from "../src/components/Navbar";
import Image from "../src/components/Image";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import { configure, shallow } from "enzyme";
// import Adapter from "enzyme-adapter-react-16";
// import toJson from "enzyme-to-json";

it("renders correctly", () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("render correctly Navbar component", () => {
  const component = renderer.create(<Navbar />).toJSON();
  expect(component).toMatchSnapshot();
});

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe("<Image />", () => {
  let shallowComp;
  beforeEach(() => {
    const props = {
      key: 1,
      url:
        "https://pixabay.com/get/52e5d3414e55a414f6da8c7dda79367d1c3fdcec53536c4870287add9248c05cb8_1280.jpg"
    };

    shallowComp = shallow(<Image url={props.url} />);
  });
  test("should render 3 imgs- link, dots and share", () => {
    const imgResult = shallowComp.find("img");
    expect(imgResult).toHaveLength(3);
  });
});

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <BrowserRouter>
      <Image />
    </BrowserRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

describe("<Navbar/>", () => {
  let shallowComp;
  beforeEach(() => {
    shallowComp = shallow(<Navbar />);
  });

  test("should have 8", () => {
    const button = shallowComp.find("button");
    expect(button).toHaveLength(8);
  });
});
