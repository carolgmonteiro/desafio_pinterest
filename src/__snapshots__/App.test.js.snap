// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly Navbar component 1`] = `
<div
  className="btn-box"
>
  <button
    className="logo-pinterest"
    id="btn-pinterest"
  >
    <img
      alt="logo-pinterest"
      src={
        Object {
          "ReactComponent": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
          "__esModule": true,
          "default": "logo.svg",
        }
      }
    />
  </button>
  <div
    className="search-box-content"
  >
    <div
      className="search-box"
    >
      <button
        className="btn-search"
      >
        <img
          alt="search-button"
          src={
            Object {
              "ReactComponent": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
              "__esModule": true,
              "default": "search.svg",
            }
          }
        />
      </button>
      <input
        placeholder="Search"
        type="text"
      />
    </div>
    <div
      className="button-content"
    >
      <div
        className="button-box"
      >
        <button
          className="btn-nav-name"
        >
          Home
        </button>
        <button
          className="btn-nav-name"
        >
          Following
        </button>
        <button
          className="btn-nav-name"
        >
          <p
            className="user-icon"
          >
            C
          </p>
          <p
            className="username"
          >
            Carolina
          </p>
        </button>
      </div>
      <div
        className="border"
      />
      <button
        className="btn-nav"
      >
        <img
          alt="chat-button"
          className="icon"
          src={
            Object {
              "ReactComponent": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
              "__esModule": true,
              "default": "chat.svg",
            }
          }
        />
      </button>
      <button
        className="btn-nav"
      >
        <img
          alt="notifications-button"
          className="icon"
          src={
            Object {
              "ReactComponent": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
              "__esModule": true,
              "default": "bell.svg",
            }
          }
        />
      </button>
      <button
        className="btn-nav"
      >
        <img
          alt="settings-button"
          className="icon"
          src={
            Object {
              "ReactComponent": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
              "__esModule": true,
              "default": "dots.svg",
            }
          }
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`renders correctly 1`] = `
Array [
  <nav
    className="header-content"
    id="navbar"
  >
    <div
      className="btn-box"
    >
      <button
        className="logo-pinterest"
        id="btn-pinterest"
      >
        <img
          alt="logo-pinterest"
          src={
            Object {
              "ReactComponent": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
              "__esModule": true,
              "default": "logo.svg",
            }
          }
        />
      </button>
      <div
        className="search-box-content"
      >
        <div
          className="search-box"
        >
          <button
            className="btn-search"
          >
            <img
              alt="search-button"
              src={
                Object {
                  "ReactComponent": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "render": [Function],
                  },
                  "__esModule": true,
                  "default": "search.svg",
                }
              }
            />
          </button>
          <input
            placeholder="Search"
            type="text"
          />
        </div>
        <div
          className="button-content"
        >
          <div
            className="button-box"
          >
            <button
              className="btn-nav-name"
            >
              Home
            </button>
            <button
              className="btn-nav-name"
            >
              Following
            </button>
            <button
              className="btn-nav-name"
            >
              <p
                className="user-icon"
              >
                C
              </p>
              <p
                className="username"
              >
                Carolina
              </p>
            </button>
          </div>
          <div
            className="border"
          />
          <button
            className="btn-nav"
          >
            <img
              alt="chat-button"
              className="icon"
              src={
                Object {
                  "ReactComponent": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "render": [Function],
                  },
                  "__esModule": true,
                  "default": "chat.svg",
                }
              }
            />
          </button>
          <button
            className="btn-nav"
          >
            <img
              alt="notifications-button"
              className="icon"
              src={
                Object {
                  "ReactComponent": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "render": [Function],
                  },
                  "__esModule": true,
                  "default": "bell.svg",
                }
              }
            />
          </button>
          <button
            className="btn-nav"
          >
            <img
              alt="settings-button"
              className="icon"
              src={
                Object {
                  "ReactComponent": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "render": [Function],
                  },
                  "__esModule": true,
                  "default": "dots.svg",
                }
              }
            />
          </button>
        </div>
      </div>
    </div>
    <hr />
  </nav>,
  <div
    className="images-container"
  >
    <div
      style={Object {}}
    >
      <div
        className="infinite-scroll-component "
        style={
          Object {
            "WebkitOverflowScrolling": "touch",
            "height": "auto",
            "overflow": "auto",
          }
        }
      >
        <div
          className="my-masonry-grid"
        />
        <h4>
          Loading...
        </h4>
      </div>
    </div>
  </div>,
]
`;
